### PR TITLE
[NFC] Drop SILLocation::LocationKind::NoneKind

### PR DESCRIPTION
**Pull request by CodaFi**
Its last real usage was removed in
c06bd833586055ab75c166b2fe8731280d8c4c41 and a series of subsequent
commits that cleared out some old assertions.

Noticed by inspection.

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Review by gottesmm — COMMENTED** [review]
SGTM if there are no other uses. I would ask @adrian-prantl to confirm this is ok. My feeling is that the right way to represent None is with an optional. With that being said, would it make sense to renumber the cases as a result of this?

**Comment by CodaFi** [comment]
> My feeling is that the right way to represent None is with an optional.

None was used a long time ago to represent artificial locations, but there's an explicit kind for that now.

> would it make sense to renumber the cases as a result of this?

Take it or leave it, really.  It's being packed into a 3-bit field right now so there isn't a space savings here yet.

File: include/swift/SIL/SILLocation.h [hunk 1]
```diff
@@ -69,8 +69,6 @@ class SILLocation {
 
 public:
   enum LocationKind : unsigned {
-    // FIXME: NoneKind is to be removed.
-    NoneKind = 0,
     RegularKind = 1,
     ReturnKind = 2,
     ImplicitReturnKind = 3,
```
Comment (adrian-prantl) [inline include/swift/SIL/SILLocation.h]: At this point I would just let RegularKind start at zero.
Also, can you make sure the validation tests pass? I could imagine this being accidentally during by SIL deserialization.

**Review by adrian-prantl — APPROVED** [review]

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
Withdrawing this.  It became a refactoring job...

**Pull request closed without merging**